backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponent.java

```diff
@@ -52,9 +55,18 @@ public Element render() {
         DiagramRenderingCache cache = new DiagramRenderingCache();
 
         // @formatter:off
+        List<Node> allPrevNodes = this.props.getPreviousDiagram()
+                .map(Diagram::getNodes)
```

Comment:
This is only the top-level nodes. Is it expected? If yes, maybe `topLevelNodes` would be a better name.